From 1f713320fe3a47881e31b9ea7287b159ef9a77c0 Mon Sep 17 00:00:00 2001
From: Yu-hsin Wang <yuhsingw@google.com>
Date: Thu, 28 Apr 2022 17:02:44 +0800
Subject: [PATCH 590/757] dev: Expose ResetRequestPort constructor

Port::Port is in protected scope. ResetRequestPort should expose the
constructor by itself.

Change-Id: I72ce701fca89379f90e212d7411f481ae1e1977a
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/59209
Tested-by: kokoro <noreply+kokoro@google.com>
Maintainer: Gabe Black <gabe.black@gmail.com>
Reviewed-by: Jason Lowe-Power <power.jg@gmail.com>
Reviewed-by: Gabe Black <gabe.black@gmail.com>
---
 src/dev/reset_port.hh | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/dev/reset_port.hh b/src/dev/reset_port.hh
index 2846900..a08db1c 100644
--- a/src/dev/reset_port.hh
+++ b/src/dev/reset_port.hh
@@ -57,7 +57,8 @@ class ResetResponsePort : public ResetResponsePortBase
 class ResetRequestPort : public Port
 {
   public:
-    using Port::Port;
+    ResetRequestPort(const std::string &_name, PortID _id)
+        : Port(_name, _id) {}
     void bind(Port &p) override;
     void unbind() override;
     void requestReset();
-- 
1.8.3.1

